PhysicsTools/IsolationAlgos/plugins/CandIsolatorFromDeposits.cc

```diff
@@ -60,11 +60,20 @@ using namespace edm;
 using namespace reco;
 using namespace reco::isodeposit;
 
-bool isNumber(const std::string &str) {
-  static const std::regex re("^[+-]?(\\d+\\.?|\\d*\\.\\d*)$");
-  return regex_match(str.c_str(), re);
-}
-double toNumber(const std::string &str) { return atof(str.c_str()); }
+namespace {
+  bool toNumber(const std::string &str, double &result) {
+    if (str.empty()) {
+      return false;
+    }
+    const char *first = str.c_str();
+    const char *last = first + str.size();
+    if (*first == '+') {
+      ++first;
+    }
+    const auto [ptr, ec] = std::from_chars(first, last, result);
+    return ec == std::errc();
```

Comment:
```suggestion
    return ec == std::errc() and ptr == last;
```
I think the condition signifying "is a number" should be stricter in this case, because of the specifics of this plugin.

Looks like `weight` is a `cms.string` in the PSet in order to decide at runtime if the "weight" is (1) just a number, or (2) an expression/function.

With this PR as is, `weight = cms.string("45 + 3")` (or, `"45 + 4*pt"`) would be translated internally to `weight_ == 45.` and `usesFunction_ == false`, which is not the intended behaviour (afaiu). Test code is [here](https://gist.github.com/missirol/81935e0675250905ac4f30e9806b3255).

PS. Regarding

> A potential downside is the larger amount of code.

I guess (didn't test) one way to reduce code is to just rely on `weightExpr_` regardless (removing `weight_` and `usesFunction_`), but maybe this is significantly less efficient in cases like `weight = cms.string("45")` (?).